### PR TITLE
refactor(auth): make authentication code more clear

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

File: huaweicloud/config/auth.go [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+				providerConfig = v
+				break
+			}
+		}
+		if (providerConfig == Profile{}) {
+			return fmt.Errorf("Error finding profile %s from shared config file", current)
+		}
+	} else {
+		return fmt.Errorf("The specified shared config file %s does not exist", profilePath)
+	}
```
Comment (Lance52259) [inline huaweicloud/config/auth.go]: Suggest to put the `else` logic in the `if`, and extract the logic in the `if` (means isExist()).

Comment (niuzhenguo) [reply huaweicloud/config/auth.go]: done

File: huaweicloud/config/auth.go [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+			DomainID:         c.DomainID,
+			Domain:           c.DomainName,
+			AgencyName:       c.AgencyName,
+			AgencyDomainName: c.AgencyDomainName,
+		}
+	} else {
+		pao = golangsdk.AKSKAuthOptions{
+			BssDomainID: c.DomainID,
+			BssDomain:   c.DomainName,
+			ProjectName: c.TenantName,
```
Comment (hmbig2) [inline huaweicloud/config/auth.go:180]: can we change projectName->TenantName, ProjectId->TenantID,
so this block in buildClientByxxx will be the same

Comment (hmbig2) [inline huaweicloud/config/auth.go:180]: can we change projectName->TenantName, ProjectId->TenantID,
so this block in buildClientByxxx will be the same

File: huaweicloud/config/auth.go [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+
+	client, err = genClient(c, dao)
+	if err == nil {
+		c.DomainClient = client
+	}
+	return err
+}
+
+func buildClientByToken(c *Config) error {
+	var pao, dao golangsdk.AuthOptions
```
Comment (Jason-Zhang9309) [inline huaweicloud/config/auth.go]: suggest replace 'pao' and 'dao' with more meaningful names

Comment (Jason-Zhang9309) [inline huaweicloud/config/auth.go:123]: suggest replace 'pao' and 'dao' with more meaningful names

File: huaweicloud/config/auth.go [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+			Domain:           c.DomainName,
+			AgencyName:       c.AgencyName,
+			AgencyDomainName: c.AgencyDomainName,
+		}
+	} else {
+		pao = golangsdk.AKSKAuthOptions{
+			BssDomainID: c.DomainID,
+			BssDomain:   c.DomainName,
+			ProjectName: c.TenantName,
+			ProjectId:   c.TenantID,
```
Comment (hmbig2) [inline huaweicloud/config/auth.go:222]: can we change this block to : golangsdk.AuthOptions{
			DomainID:   c.DomainID,
			DomainName: c.DomainName,
			TenantID:   c.TenantID,
			TenantName: c.TenantName,
		}
so this block in buildClientByxxx will be the same

Comment (hmbig2) [inline huaweicloud/config/auth.go:181]: can we change this block to : golangsdk.AuthOptions{
			DomainID:   c.DomainID,
			DomainName: c.DomainName,
			TenantID:   c.TenantID,
			TenantName: c.TenantName,
		}
so this block in buildClientByxxx will be the same

File: huaweicloud/config/auth.go [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+func genClient(c *Config, ao golangsdk.AuthOptionsProvider) (*golangsdk.ProviderClient, error) {
+	client, err := huaweisdk.NewClient(ao.GetIdentityEndpoint())
+	if err != nil {
+		return nil, err
+	}
+
+	// Set UserAgent
+	client.UserAgent.Prepend("terraform-provider-iac")
+
+	config, err := generateTLSConfig(c)
```
Comment (ShiChangkuo) [inline huaweicloud/config/auth.go:110]: suggest to move **generateTLSConfig** into auth.go

File: huaweicloud/config/config.go [hunk 6]
```diff
@@ -104,38 +69,22 @@ type Config struct {
...
+	Password         string
+	AgencyName       string
+	AgencyDomainName string
+	DelegatedProject string
 }
 
 func (c *Config) LoadAndValidate() error {
 	if c.MaxRetries < 0 {
-		return fmt.Errorf("max_retries should be a positive value")
+		return fmt.Errorf("max_retries should be a positive value.")
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go]: go-lint: error strings should not be capitalized or end with punctuation or a newline

File: huaweicloud/config/auth.go (hunk 3 above)
Comment (niuzhenguo) [reply huaweicloud/config/auth.go]: done

File: huaweicloud/config/auth.go (hunk 4 above)
Comment (niuzhenguo) [reply huaweicloud/config/auth.go:222]: done

File: huaweicloud/config/auth.go (hunk 5 above)
Comment (niuzhenguo) [reply huaweicloud/config/auth.go:110]: done

File: huaweicloud/config/config.go (hunk 6 above)
Comment (niuzhenguo) [reply huaweicloud/config/config.go]: done

**Pull request merged**